cpp/cmake/Modules/ConfigureCUDA.cmake

```diff
@@ -43,5 +43,16 @@ endif()
 # Debug options
 if(CMAKE_BUILD_TYPE MATCHES Debug)
   message(VERBOSE "CUDF: Building with debugging flags")
+  list(APPEND CUDF_CXX_FLAGS -rdynamic)
   list(APPEND CUDF_CUDA_FLAGS -Xcompiler=-rdynamic)
 endif()
+
+if(CUDF_BUILD_STACKTRACE_DEBUG)
+  message(VERBOSE "CUDF: Building with flags '-rdynamic -Og -NDEBUG'")
+  set(CMAKE_CXX_FLAGS_RELEASE "") # Remove the current optimization level
+  set(CMAKE_CUDA_FLAGS_RELEASE "") # Remove the current optimization level
+  list(APPEND CUDF_CXX_FLAGS -rdynamic -Og)
+  list(APPEND CUDF_CUDA_FLAGS -Xcompiler=-rdynamic -Xcompiler=-Og)
+  list(APPEND CUDF_CXX_DEFINITIONS NDEBUG)
```

Comment:
`NDEBUG` is already defined when in release mode via the default flags, lets not go with this approach as it also has problem when a user does a debug build as it will remove asserts from working.